src/OnlineSales/Services/EmailFromTemplateService.cs

```diff
@@ -51,11 +65,31 @@ private async Task<EmailTemplate> GetEmailTemplate(string name, int contactId)
         {
             var contactLanguage = pgDbContext.Contacts!.FirstOrDefault(c => c.Id == contactId)!.Language;
 
-            var template = await pgDbContext.EmailTemplates!.FirstOrDefaultAsync(x => x.Name == name && x.Language == contactLanguage);
+            var template = await GetEmailTemplateByLanguage(name, contactLanguage);
 
             return template!;
         }
 
+        private async Task<EmailTemplate?> GetEmailTemplateByLanguage(string name, string? language)
+        {
+            string defLang = config.Language!;
+
+            // set default if not set
+            language ??= defLang;
+
+            // try find template by provided language
+            // with 2 and 5 language codes
+            var template = await pgDbContext.EmailTemplates!.FirstOrDefaultAsync(x => x.Name == name && (x.Language.Length == 2 ? x.Language == language.Substring(0, 2) : x.Language == language));
```

Comment:
@Mixer57 x.Language.Length should never be 2 letters, always 5. If it is not the case we need to fix it on the database level now.